lib/ReportHistoryStore.jsx

```diff
@@ -202,6 +202,35 @@ export default class ReportHistoryStore {
         this.cache[reportID] = newCache.sort((a, b) => b.reportActionTimestamp - a.reportActionTimestamp);
     }
 
+    /**
+     * Merges history items using the reportActionID into the cache and creates it if it doesn't yet exist.
+     * We then sorts them by timestamp and then reprotActionID to ensure a consistent order if two report actions have the same timestamp
+     *
+     * @param {Number} reportID
+     * @param {Object[]} newHistory
+     */
+    mergeHistoryByReportActionID(reportID, newHistory) {
+        if (newHistory.length === 0) {
+            return;
+        }
+
+        const newCache = newHistory.reverse().reduce((prev, curr) => {
+            if (!prev.some((item) => item.reportActionID === curr.reportActionID)) {
+                prev.unshift(curr);
+            }
+            return prev;
+        }, this.cache[reportID] || []);
+
+        // Sort items in case they have become out of sync
+        this.cache[reportID] = newCache.sort((a, b) => {
+            if (b.reportActionTimestamp !== a.reportActionTimestamp) {
+                return b.reportActionTimestamp - a.reportActionTimestamp;
+            }
+
+            return a.reportActionID.localeCompare(b.reportActionID);
```

Comment:
I also don't get this, if they are fully equal then we call localCompare? Why? Wouldn't it always return 0?

Reply:
@iwiznia this is sorting by both `reportActionTimestamp` and then reportActionID, the previous comparison is for timestamps

Reply:
Ohhh, I totally missed that! Makes sense then.